### PR TITLE
Fix table styling bug

### PR DESCRIPTION
**Pull request by jcmustin**
[This PR](https://github.com/ForumMagnum/ForumMagnum/pull/3558) added a `display: table` style to figure elements that wrap table elements, in order to preserve the appearance of tables from the editor view. However, this also results in a table as a direct descendant of another table, which is invalid HTML. This causes unexpected behavior on Safari. (~~Also it seems like a total coincidence that it happens to look like the editor view the rest of the time~~ edit: nevermind, CKEditor does in fact use `display: table` in the editor view — I claim that they shouldn't.) This PR replaces `display: table` with `width: fit-content`, which in combination with pre-existing styles is sufficient to capture the appearance of CKEditor's table widget.

Fixes https://github.com/ForumMagnum/ForumMagnum/issues/6321

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203572531350839) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by s-cheng** [comment]
@jcmustin Looks like it fixed the width issue, but there's an issue with the height now:

Screen Shot 2022-12-19 at 1 12 02 PM

File: packages/lesswrong/themes/stylePiping.ts [hunk 1]
```diff
@@ -256,7 +256,17 @@ const baseBodyStyles = (theme: ThemeType): JssStyles => ({
...
+    height: 'fit-content !important',
+  },
+  // Many column tables should overflow instead of squishing
+  //  - NB: As of Jan 2023, this does not work on firefox, so ff users will have
+  //    squishy tables (which is the default behavior above)
+  '& figure.table:has(> table > tbody > tr > td + td + td + td)': {
+    overflowX: 'auto',
+    '& table': {
+      width: 700,
+    },
```
Comment (jpaddison3) [inline packages/lesswrong/themes/stylePiping.ts:269]: Very happy with this one. My (and Jonathan's) first use of a [parent selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) and it's great.

**Review by jpaddison3 — APPROVED** [review]
LGTM

**Comment by jpaddison3** [comment]
(Tested on safari mobile as well.)

**Pull request merged**